### PR TITLE
[Settings]: Added change password view

### LINKED ISSUE #10037: Re-encrypt db screen
When re-encryption of the db happens, it may take a lot of time depending on db size, we needs screens for this

Flows when re-encryption happens:
migrate profile to keycard
convert keycard user to regular keycard
change password

@alaibe check if there is a indicator of progress
@John-44 @benjthayer design the screens

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #10037

Biometrics option to be added here: https://github.com/status-im/status-desktop/issues/13302

Depends on nimqml PR:
https://github.com/status-im/nimqml/pull/56

### What does the PR do
Settings: Added change password view

1. Add support to restart the app
2. Move change password call to the thread pool
3. Add view configuration options to `PasswordView`
4. Create the new change password settings screen

### Affected areas
Settings
Onboarding
Keycard migration popup

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/47811206/95ac9f78-71f8-491b-9d78-09e9a185047a

### CODE REVIEW

**Review by caybro — APPROVED** [review]
LGTM

Modulo the e2e failures that are likely due to the offset in `Constants.settingsSubsection`

**Comment by caybro** [comment]
@alexjba where's the related nimqml PR? :)

File: ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+import StatusQ.Core 0.1
+import StatusQ.Popups 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+import "../views"
+
+StatusModal {
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml:21]: Pls use `StatusDialog` for new code

File: ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                border.width: !d.passwordChanged ? 1 : 0
+                border.color: Theme.palette.baseColor5
+                color: d.passwordChanged ? "transparent" : bgColor
+            }
+        }
+    }
+
+    leftButtons: [
+        StatusFlatButton {
+            text: "Cancel"
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml:105]:
```suggestion
            text: qsTr("Cancel")
```

**Comment by anastasiyaig** [comment]
Sure, feel free to merge the PR as is, i will address new changes in new tests later today or tomorrow

File: vendor/DOtherSide/lib/src/DOtherSide.cpp [hunk 3]
```diff
@@ -371,6 +371,12 @@ void dos_qguiapplication_quit()
     QMetaObject::invokeMethod(qGuiApp, "quit", Qt::QueuedConnection);
 }
 
+void dos_qguiapplication_restart()
```
Comment (micieslak) [inline vendor/DOtherSide/lib/src/DOtherSide.cpp:374]: I wonder if for things like that we need to go via nim layer. What about creating independent utility in cpp and expose it e.g. as a singleton to qml?

File: storybook/pages/ConfirmChangePasswordModalPage.qml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                    onTriggered: {
+                        if (successFlow.checked) {
+                            modal.passwordSuccessfulyChanged()
+                        } else {
+                            modal.close()
+                        }
+                    }
+                }
+            }
+        }
```
Comment (micieslak) [inline storybook/pages/ConfirmChangePasswordModalPage.qml:73]:
```suggestion
        ConfirmChangePasswordModal {
            id: modal

            visible: true
            modal: false
            closePolicy: Popup.NoAutoClose

            onChangePasswordRequested: {
                passwordChangedTimer.start()
            }

            Component.onCompleted: {
                popupBg.popupIntance = modal
            }

            Timer {
                id: passwordChangedTimer
                interval: 2000
                repeat: false
                onTriggered: {
                    if (successFlow.checked) {
                        modal.passwordSuccessfulyChanged()
                    } else {
                        modal.close()
                    }
                }
            }
        }
```
and only `open` call in "reopen" button :)

File: ui/app/AppLayouts/Profile/views/PasswordView.qml
Comment (micieslak) [inline ui/app/AppLayouts/Profile/views/PasswordView.qml:1]: There are two file `PasswordView` with some code duplicated. I think it's good chance to clean it up a bit.

**Review by micieslak — COMMENTED** [review]
LGTM in general, just some questions and doubts

Is this behavior correct? I mean red bar filling the whole width:
[Kazam_screencast_00090.webm](https://github.com/status-im/status-desktop/assets/20650004/e2688405-29cc-4760-8752-20b873876351)

File: vendor/DOtherSide/lib/src/DOtherSide.cpp (hunk 3 above)
Comment (alexjba) [reply vendor/DOtherSide/lib/src/DOtherSide.cpp:374]: I've had some thoughts about it. That was my first intention as well, but having the implementation in DOtherSide next to the `quit` implementation felt more intuitive now. As a result, the qml access needed to go needed to go through `utils.nim`. It's the place where I'd search for such an implementation (or anything else related to qGuiApp).

But we could add a task to move the entire utils.nim to cpp. WDYT? I think it's feasible.

File: ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml (hunk 1 above)
Comment (alexjba) [reply ui/app/AppLayouts/Profile/popups/ConfirmChangePasswordModal.qml]: Fixed

File: ui/app/AppLayouts/Profile/views/PasswordView.qml
Comment (alexjba) [reply ui/app/AppLayouts/Profile/views/PasswordView.qml:1]: Fixed

File: vendor/DOtherSide/lib/src/DOtherSide.cpp (hunk 3 above)
Comment (micieslak) [reply vendor/DOtherSide/lib/src/DOtherSide.cpp:374]: Yeah, that's good idea I think!

**Comment by micieslak** [comment]
> > Is this behavior correct? I mean red bar filling the whole width: [Kazam_screencast_00090.webm](https://github.com/status-im/status-desktop/assets/20650004/e2688405-29cc-4760-8752-20b873876351)
> 
> @micieslak Not sure I follow. The bar width has always been the same as the input field (even in the old designs). Here's the current design, maybe it helps: https://www.figma.com/file/d0G7m8X6ELjQlFOEKQpn1g/Profile-WIP?type=design&node-id=11-109778&mode=design&t=XpXPtlra05AfWP3o-0

After thinking more about it, I came to the conclusion that the current behavior is ok. The length of the bar just reflect length of the password.

**Review by micieslak — APPROVED** [review]

**Pull request merged**